--- origsrc/mutter-3.14.4/doc/reference/Makefile.am	2015-03-23 11:49:32.000000000 -0500
+++ src/mutter-3.14.4/doc/reference/Makefile.am	2015-07-16 17:48:04.136258500 -0500
@@ -118,6 +118,12 @@ IGNORE_HFILES += \
 	$(NULL)
 endif
 
+if !HAVE_NATIVE_BACKEND
+IGNORE_HFILES += \
+	native				\
+	$(NULL)
+endif
+
 MKDB_OPTIONS+=--ignore-files="$(IGNORE_HFILES)"
 
 # Images to copy into HTML directory.
--- origsrc/mutter-3.14.4/src/backends/meta-cursor-private.h	2015-03-23 11:49:32.000000000 -0500
+++ src/mutter-3.14.4/src/backends/meta-cursor-private.h	2015-07-16 17:48:04.142759300 -0500
@@ -25,7 +25,9 @@
 #include "meta-cursor.h"
 
 #include <cogl/cogl.h>
+#ifdef HAVE_NATIVE_BACKEND
 #include <gbm.h>
+#endif
 
 typedef struct {
   CoglTexture2D *texture;
--- origsrc/mutter-3.14.4/src/backends/meta-cursor-renderer.h	2015-03-23 11:49:32.000000000 -0500
+++ src/mutter-3.14.4/src/backends/meta-cursor-renderer.h	2015-07-16 17:48:04.145759700 -0500
@@ -30,7 +30,9 @@
 #include <meta/screen.h>
 #include "meta-cursor.h"
 
+#ifdef HAVE_NATIVE_BACKEND
 #include <gbm.h>
+#endif
 
 #define META_TYPE_CURSOR_RENDERER            (meta_cursor_renderer_get_type ())
 #define META_CURSOR_RENDERER(obj)            (G_TYPE_CHECK_INSTANCE_CAST ((obj), META_TYPE_CURSOR_RENDERER, MetaCursorRenderer))
